ros2interface/ros2interface/verb/show.py

```diff
@@ -43,12 +242,105 @@ def add_arguments(self, parser, cli_name):
             help="Show an interface definition (e.g. 'example_interfaces/msg/String'). "
                  "Passing '-' reads the argument from stdin (e.g. "
                  "'ros2 topic type /chatter | ros2 interface show -').")
+        parser.add_argument("-r", "--raw",
+                          dest="raw", default=False, action="store_true",
+                          help="show raw message text, including comments")
         arg.completer = type_completer
 
     def main(self, *, args):
-        try:
-            file_path = get_interface_path(args.type)
-        except LookupError as e:
-            return str(e)
+
+        InterfaceExpander(args.type).print(raw=args.raw)
```

Comment:
It looks like this is parsing the legacy message definition. I think this should reuse the legacy message parsing code in `rosidl`. This lives in a package called [`rosidl_adapter`](https://github.com/ros2/rosidl/tree/master/rosidl_adapter). In particular I think the methods [`parse_message_string`](https://github.com/ros2/rosidl/blob/b4a4431b49f0aa786f9cf990acf2bcccc5d3c98a/rosidl_adapter/rosidl_adapter/parser.py#L444), [`parse_service_string`](https://github.com/ros2/rosidl/blob/b4a4431b49f0aa786f9cf990acf2bcccc5d3c98a/rosidl_adapter/rosidl_adapter/parser.py#L822), and [`parse_action_string()`](https://github.com/ros2/rosidl/blob/b4a4431b49f0aa786f9cf990acf2bcccc5d3c98a/rosidl_adapter/rosidl_adapter/parser.py#L866) could be reused here depending on the type of message given in.

Those give `ActionSpecification`, `ServiceSpecification`, and `MessageSpecification` instances. The most important one here is the `MessageSpecification`. I think if `raw` is `False` then this code should use those functions to parse the legacy definition, modify the `MessageSpecification` instance (say by removing comments), then finally calling [`str(msg_spec_instance)`](https://github.com/ros2/rosidl/blob/b4a4431b49f0aa786f9cf990acf2bcccc5d3c98a/rosidl_adapter/rosidl_adapter/parser.py#L424) to write out the result.